### PR TITLE
UI: simplify layout calculation

### PR DESCRIPTION
**Pull request by deanlee**
1.add Rect struct
2.remove many layout variables
3.detect screen width&height dynamically
3.use a single viz_rect  for responsive layout drawing

after refactor:
ui_viz_rx -> viz_rect.x
box_y -> viz_rect.y
ui_viz_rw -> viz_rect.w
box_h -> viz_rect.h
vwp_w -> s->fb_w
vwp_h -> s->fb_h

### CODE REVIEW

File: selfdrive/ui/ui.hpp [hunk 1]
```diff
@@ -37,36 +37,20 @@
...
-const int vwp_w = 2160;
-#else
-const int vwp_w = 1920;
-#endif
+typedef struct Rect {
+  int x, y, w, h;
+  int centerX() const {return x+w/2;}
+  int right() const {return x+w;}
+  int bottom() const {return y+h;}
+  bool ptInRect(int px, int py) const{ return px>=x&&px<(x+w)&&py>=y&&py<(y+h); }
```
Comment (adeebshihadeh) [inline selfdrive/ui/ui.hpp:45]: add some whitespace and/or use another line

Comment (deanlee) [reply selfdrive/ui/ui.hpp]: done

File: selfdrive/ui/qt/window.hpp [hunk 2]
```diff
@@ -25,7 +25,8 @@ public slots:
 
 };
 
-
+const int vwp_w = 1920;
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/window.hpp:28]: this isn't right for QCOM2

Comment (deanlee) [reply selfdrive/ui/qt/window.hpp]: fixed

Comment (adeebshihadeh) [reply selfdrive/ui/qt/window.hpp]: should be 1920 if not QCOM2

Comment (deanlee) [reply selfdrive/ui/qt/window.hpp:32]: done

**Comment by adeebshihadeh** [comment]
Nice cleanup, thanks @deanlee!

**Pull request merged**